index.js

```diff
@@ -0,0 +1,126 @@
+   * to generate the key to store the value returned by the valFn under.
+   * @property {String} prefix The prefix to namespace the event under.
+   * @property {Number} ttl How long the event should live in Redis before
+   *     expiring (in seconds).
+   * @property {Function} idFn A function for extracting an event's ID.
+   * @property {Function} valFn A function for extracting the value to store
+   *    for an event in Redis.
+   * @property {Object} eve The event to mark as seen.
+   * @property {Function} done Node style callback.
+   */
+  markAsSeen(prefix, ttl, idFn, valFn, eve, done) {
+    const eventId = idFn(eve);
+    // Safety belts.
+    if (!eventId) return done();
+
+    return this._redisConnection.setex(`${prefix}:${eventId}`, ttl, valFn(eve), done);
+  }
+
+  /**
+   * @typedef EventHandler
+   * @property {String} prefix The prefix to use to namespace events.
+   * @property {Function} timestampFn A function for extracting when an event
+   *    occurred.
+   * @property {Function} idFn A function for extracting an event's ID.
+   * @property {Function} valFn A function for extracting the value to store
+   *    for an event in Redis.
+   * @property {Number} window The number of milliseconds before now, within
+   *    which we should consider events.
+   * @property {Number} ttl How long an event should remain in Redis before
+   *    expiring.
+   */
+
+  /**
+   * Registers the given handler for the given type.
+   * @param {String} type The event type to register the handler for.
+   * @param {EventHandler} handler The event handler to register.
+   * @throws Error If there was already a handler registered for the given type.
+   */
+  registerHandler(type, handler) {
+    if (this._handlers[type]) throw new Error('cannot overwrite existing handler');
+
+    // Ensure they our required properties exist on the handler.
+    assert(handler.prefix, 'Handler must specify a prefix to namespace events.');
+    assert(handler.timestampFn, 'Handler must specify a timestampFn.');
+    assert(handler.idFn, 'Handler must specify an idFn.');
+    assert(handler.valFn, 'Handler must specify a valFn.');
+    assert(handler.window, 'Handler must specify a window.');
+    
+    // Be nice, if the user didn't specify the ttl on the handler, set it for
+    // them from the window.
+    if (!handler.ttl) handler.ttl = Math.floor(handler.window / 1000);
+
+    this._handlers[type] = handler;
+  }
+
+  /**
+   * Checks if this is the first time we've seen an event of the given type. If
+   * it is, we mark the event as seen and return true, otherwise we return
+   * false. If the event occurred outside our window of consideration, we also
+   * return false.
+   * @param {String} type The type of the event that we're inspecting.
+   * @param {Object} eve The event we're inspecting.
+   * @param {Function} done Node style callback.
+   */
+  inspectEvent(type, eve, done) {
```

Comment:
I envision this being the primary usage of this library, the others are provided purely to give other users access to the primitives if they so desire. It seems to be causing confusion though, so I'll just rip them out.